projects/hello_world/src/hello_world.c

```diff
@@ -0,0 +1,35 @@
+//
+// Created by henry on 2021-01-09.
+//
+#include <stdint.h>
+
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+
+typedef struct Counters {
+  uint8_t counter_a;
+  uint8_t counter_b;
+} Counters;
+void callback(SoftTimerId id, void *context) {
+  static bool b_increment = false;
```

Comment:
A function-local static variable is a clever way to get around the "no global variables" requirement, but we generally don't use them unless there's a *really good reason*. Instead, consider using whether `counter_a` is even or odd to decide whether to increment `counter_b`.